src/algebra/monoid_algebra/basic.lean

```diff
@@ -1503,6 +1529,20 @@ finset.induction_on s rfl $ λ a s has ih, by rw [prod_insert has, ih,
 
 end
 
+lemma map_domain_algebra_map (k A : Type*) {H F : Type*} [comm_semiring k] [semiring A]
+  [algebra k A] [add_monoid G] [add_monoid H] [add_monoid_hom_class F G H] (f : F) (r : k) :
+  map_domain f (algebra_map k (add_monoid_algebra A G) r) =
+    (algebra_map k (add_monoid_algebra A H)) r :=
+by simp only [function.comp_app, map_domain_single, add_monoid_algebra.coe_algebra_map, map_zero]
+
+/--  An additive homomorphism `f : G →+ H` induces a `k`-algebra homomorphism
+`map_domain_alg_hom k A f : add_monoid_algebra A G →ₐ[k] add_monoid_algebra A H`. -/
+@[simps] def map_domain_alg_hom (k A : Type*) [comm_semiring k] [semiring A] [algebra k A]
+  [add_monoid G] {H F : Type*} [add_monoid H] [add_monoid_hom_class F G H] (f : F) :
+  add_monoid_algebra A G →ₐ[k] add_monoid_algebra A H :=
+{ commutes' := λ r, map_domain_algebra_map k A f r,
```

Comment:
Maybe just write `{ commutes' := by simp, .. map_domain_ring_hom A f }` and remove the lemma above? Or you can just write `commutes' := map_domain_algebra_map k A f,`

Reply:
Actually, I extracted the lemma as a `simp`, since Eric said he preferred it like this.  Maybe, as the lemma is not `simp` the older way was better.  I'll wait for Eric's comment on this!

Reply:
I think the lemma is worth having separately, but I agree with the
```suggestion
{ commutes' := map_domain_algebra_map k A f,
```
suggestion

Reply:
Done!  Also for the multiplicative version.

Reply:
I don't see the change - have you not pushed?

Reply:
It should now: I was commenting on the PR, while making the changes on my local version and wanted to get to the end of the suggestions before pushing.

I *think* that I have implemented everything and I certainly have pushed now!